src/data/transform.ts

```diff
@@ -641,5 +651,52 @@ export const convertToExcalidrawElements = (
       }
     }
   }
+
+  for (const [id, element] of elementsWithIds) {
+    if (element.type !== "frame") {
+      continue;
+    }
+    const frame = elementStore.getElement(id);
+
+    if (!frame) {
+      throw new Error(`Excalidraw element with id ${id} doesn't exist`);
+    }
+
+    let x1 = Infinity;
+    let y1 = Infinity;
+    let x2 = -Infinity;
+    let y2 = -Infinity;
+    const PADDING = 10;
+
+    element.elements.forEach((id) => {
+      const newElementId = oldToNewElementIdMap.get(id);
+
+      if (!newElementId) {
+        throw new Error(`Element with ${id} wasn't mapped correctly`);
+      }
+
+      const elementInFrame = elementStore.getElement(newElementId);
+
+      if (!elementInFrame) {
+        throw new Error(`Frame element with id ${newElementId} doesn't exist`);
+      }
+
+      x1 = Math.min(x1, elementInFrame.x);
+      y1 = Math.min(y1, elementInFrame.y);
+      x2 = Math.max(x2, elementInFrame.x + elementInFrame.width);
+      y2 = Math.max(y2, elementInFrame.y + elementInFrame.height);
+      Object.assign(elementInFrame, { frameId: frame.id });
+    });
+
+    x1 = x1 - PADDING;
+    x2 = x2 + PADDING;
+    y1 = y1 - PADDING;
+    y2 = y2 + PADDING;
+
+    const width = x2 - x1;
+    const height = y2 - y1;
+    Object.assign(frame, { x: x1, y: y1, width, height });
```

Comment:
use `getCommonBounds` instead?

Reply:
what if the frame element already has `width` & `height` set?

Reply:
> what if the frame element already has `width` & `height` set?

Didn't want to add this use case rn but since you pointed out so I am taking max of calculated and frame dimensions whichever is higher